yt/data_objects/static_output.py

```diff
@@ -654,7 +655,18 @@ def print_stats(self):
     def field_list(self):
         return self.index.field_list
 
+    @property
+    def field_info(self):
+        if self._field_info is None:
+            self.create_field_info()
+        return self._field_info
+
+    @field_info.setter
```

Comment:
Is this necessary ?
Immutability has a lot of value when dealing with parallelism, so I would advise to avoid introducing mutability where it's not needed.

Reply:
We are already doing this in far more critical places, e.g., the index. In fact, I would argue that we want some measure of immutability here.

Reply:
Sorry, I replied without really thinking this through. We don't do this with index, but we probably do want to allow this to be settable.

Reply:
I think I'm even more confused now. I don't see the connection with #5211

Reply:
I'm referring to how the tests in that PR manually alter entries in `field_info`. You wouldn't be able to do that without having the setter.

Reply:
I've come around on not having a setter for `field_info`. When it wasn't a property we allowed it to be set by anyone, but maybe it's better to be safe since we can.